pkg/reconciler/certificate/certificate_test.go

```diff
@@ -21,6 +21,7 @@ import (
 	"testing"
 	"time"
 
+	_ "knative.dev/pkg/injection/informers/kubeinformers/corev1/service/fake"
```

Comment:
Can we add a test case in `TestReconcile`?

Reply:
What kind of test exactly?

Reply:
The changes in `pkg/reconciler/certificate/certificate.go` set the http01 challenge on the kn cert status, so we can add a test case that starts with a kn cert with `knCert.Spec.DNSNames`, and a fake service in knCert's namespace and has the selector you are looking for, and verify that after reconciliation `knCert.Status.HTTP01Challenges` is set correctly